git-path/src/realpath.rs

```diff
@@ -18,15 +20,13 @@ pub(crate) mod function {
     use std::path::{Path, PathBuf};
 
     /// TODO
-    pub fn realpath(path: &Path, cwd: &Path, max_symlinks: u8) -> Result<PathBuf, Error> {
-        // TODO add test
-
+    pub fn realpath(path: impl AsRef<Path>, cwd: impl AsRef<Path>, max_symlinks: u8) -> Result<PathBuf, Error> {
```

Comment:
by doing so we accept more input types, which seems always better or is there a case where `&T` arg is preferable over` impl AsRef<T>` arg ?

Reply:
`&T` is preferable when used in `Option<&T>` as `Option<impl AsRef<T>>` makes it hard to just call the method with `None` - the compiler will argue that it doesn't know the type of the option that way, which makes sense. In that case, do `impl Into<Option<T>>` which is powerful as one can then call `f(value)` and `f(None)`.